### ISSUE TITLE
Preopen of "../" does not work (but works in wasmtime)

### ISSUE THREAD

**Issue by kanaka** [wasmerio/wasmer#372]
Using the wasi tutorial at https://github.com/CraneStation/wasmtime/blob/master/docs/WASI-tutorial.md that copies a file I see the following difference in behavior between wasmtime and wasmer:

```
$ wasmtime --dir=./ --dir=../ wasitests/copy.wasm ./test.txt ./test2.txt
$ sum test*.txt
03762     1 test.txt
03762     1 test2.txt
$ wasmtime --dir=./ --dir=../ wasitests/copy.wasm ../LICENSE ./LICENSE 
$ sum ../LICENSE LICENSE
23159    17 ../LICENSE
23159    17 LICENSE

$ rm test2.txt LICENSE 
$ wasmer run --dir=./ --dir=../ wasitests/copy.wasm ./test.txt ./test2.txt
$ sum test*.txt
03762     1 test.txt
03762     1 test2.txt
$ wasmer run --dir=./ --dir=../ wasitests/copy.wasm ../LICENSE ./LICENSE
error opening input ../LICENSE: I/O error
"RuntimeError: WebAssembly trap occured during runtime: unknown"
```

This is wasmer built from https://github.com/wasmerio/wasmer/commit/48a8391f5d199dfaf88fac7cc4eefa194a88c572 ("Fix WASI by using u32 everywhere instead of u16")

**Comment by MarkMcCaskey** [comment]
Thanks for the bug report!

Paths involving `..` haven't been implemented in `path_open` yet, we should have printed a nice error message saying that 🤔 .  I think it's because in the above commands, wasmer will be reading `LICENSE` and `test.txt` as wasm files and running them.  To fix that we can use the `--` separator, like `wasmer run --dir=./ --dir=../ wasitests/copy.wasm  -- ../LICENSE ./LICENSE`.

Changing this actually shows a new error, for some reason `../LICENSE` gets changed in to `LICENSE`, but `./../LICENSE` parses correctly and we get 
```thread 'main' panicked at 'not yet implemented: ".." in paths in `path_open` has not been implemented yet', lib/wasi/src/syscalls/mod.rs:1229:29```

I'll be looking in to the weird parsing of relative paths and implementing at least part of `..` today!

**Comment by syrusakbary** [comment]
@kanaka this should be fixed in our latest version of Wasmer 0.6.0

Closing the issue. Let us know if it doesn't fix it and we can follow with a more deep investigation.

**Comment by kanaka** [comment]
I've confirmed that this works for me now.
